compiler/types/Unify.hs

```diff
@@ -1430,7 +1430,7 @@ ty_co_match menv subst ty1 (AppCo co2 arg2) _lkco _rkco
 
 ty_co_match menv subst (TyConApp tc1 tys) (TyConAppCo _ tc2 cos) _lkco _rkco
   = ty_co_match_tc menv subst tc1 tys tc2 cos
-ty_co_match menv subst (FunTy w ty1 ty2) co _lkco _rkco
+ty_co_match menv subst (FunTy _w ty1 ty2) co _lkco _rkco
```

Comment:
Mmm… this is wrong isn't it? We should be unifying the multiplicities, shouldn't we?